packages/database/src/migrations/20210824045814-AddPacMossiRawDataDownload-modifies-data.js

```diff
@@ -0,0 +1,171 @@
+
+const getCountryDashboardItemCode = countryCode =>
+  `${countryCode}_${PROJECT_NAME}_${DASHBOARD_ITEM_NAME.split(' ').join('_')}`;
+
+const REPORT_CONFIG = {
+  surveys: [
+    {
+      code: 'PPM',
+      name: 'PacMOSSI Project Metadata',
+    },
+    {
+      code: 'PFSM',
+      name: 'PacMOSSI Field Station Mapping',
+    },
+    {
+      code: 'PAMC',
+      name: 'PacMOSSI Adult Mosquito Collection',
+    },
+    {
+      code: 'PLS',
+      name: 'PacMOSSI Larval Survey',
+    },
+    {
+      code: 'PIR',
+      name: 'PacMOSSI Insecticide Resistance',
+    },
+    {
+      code: 'PMS',
+      name: 'PacMOSSI Mosquito Storage',
+    },
+  ],
+  exportDataBuilder: {
+    dataBuilder: 'rawDataValues',
+    dataBuilderConfig: {
+      surveysConfig: {
+        PPM: {
+          entityAggregation: {
+            dataSourceEntityType: 'country',
+          },
+        },
+        PFSM: {
+          entityAggregation: {
+            dataSourceEntityType: 'field_station',
+          },
+        },
+        PAMC: {
+          entityAggregation: {
+            dataSourceEntityType: 'field_station',
+          },
+        },
+        PLS: {
+          entityAggregation: {
+            dataSourceEntityType: 'larval_habitat',
+          },
+        },
+        PIR: {
+          entityAggregation: {
+            dataSourceEntityType: 'field_station',
+          },
+        },
+        PMS: {
+          entityAggregation: {
+            dataSourceEntityType: 'district',
+          },
+        },
+      },
+    },
+  },
+};
+
+const FRONTEND_CONFIG = {
+  name: 'Download Vector Surveillance Survey Data',
+  type: 'view',
+  viewType: 'dataDownload',
+  periodGranularity: 'year',
+};
+
+const addDashboardItemToCountry = async (db, countryCode) => {
+  await insertObject(db, 'legacy_report', {
+    id: generateId(),
+    code: getCountryDashboardItemCode(countryCode),
+    data_builder: 'surveyDataExport',
+    data_builder_config: REPORT_CONFIG,
+    data_services: [{ isDataRegional: true }],
+  });
+
+  const dashboardId = generateId();
+  await insertObject(db, 'dashboard', {
+    id: dashboardId,
+    code: getCountryDashboardCode(countryCode),
+    name: DASHBOARD_NAME,
+    root_entity_code: countryCode,
+    sort_order: 2,
+  });
+
+  const dashboardItemId = generateId();
+  await insertObject(db, 'dashboard_item', {
+    id: dashboardItemId,
+    code: getCountryDashboardItemCode(countryCode),
+    config: FRONTEND_CONFIG,
+    report_code: getCountryDashboardItemCode(countryCode),
+    legacy: true,
+  });
+
+  await insertObject(db, 'dashboard_relation', {
+    id: generateId(),
+    dashboard_id: dashboardId,
+    child_id: dashboardItemId,
+    entity_types: '{country}',
+    project_codes: '{pacmossi}',
+    permission_groups: '{PacMOSSI Senior}',
+    sort_order: 2,
+  });
+};
+
+exports.up = async function (db) {
+  await Promise.all(COUNTRY_CODES.map(countryCode => addDashboardItemToCountry(db, countryCode)));
```

Comment:
```suggestion
  await Promise.all(COUNTRY_CODES.map(async countryCode => addDashboardItemToCountry(db, countryCode)));
```

Without async, sometimes it works but sometimes it won't, it would be safe to always have `async` for asynchronous functions.

Reply:
Added